lib/ask_export/survey_response_fetcher.rb

```diff
@@ -0,0 +1,118 @@
+require "faraday"
+require "json"
+
+module AskExport
+  class SurveyResponseFetcher
+    RESPONSES_PER_REQUEST = 100
+
+    def self.call(*args, &block)
+      new(*args).call(&block)
+    end
+
+    def initialize(since_time, until_time)
+      @since_time = since_time
+      @until_time = until_time
+    end
+
+    def call
+      page = 1
+      responses = []
+      loop do
+        survey_id = AskExport.config(:survey_id)
+        response = http_client.get("surveys/#{survey_id}/responses",
+                                   page: page,
+                                   page_size: RESPONSES_PER_REQUEST,
+                                   since: since_time.to_i,
+                                   until: until_time.to_i,
+                                   sort_by: "date_ended,asc",
+                                   include_labels: true,
+                                   completed: 1)
+
+        body = JSON.parse(response.body, symbolize_names: true)
+
+        responses += body.filter { |entry| entry[:status] == "completed" }
+                         .map { |entry| ResultPresenter.call(entry) }
+
+        yield responses.count if block_given?
+
+        break if body.length < RESPONSES_PER_REQUEST
+
+        # We're rate limited to 180 requests a minute so we need to slow
+        # our requests down a bit
+        sleep 0.33
+        page += 1
+      end
+
+      responses
+    end
+
+    private_class_method :new
+
+  private
+
+    attr_reader :since_time, :until_time
+
+    def http_client
+      @http_client ||= Faraday.new(
+        "https://api.smartsurvey.io/v1/",
+        params: {
+          api_token: ENV.fetch("SMART_SURVEY_API_TOKEN"),
+          api_token_secret: ENV.fetch("SMART_SURVEY_API_TOKEN_SECRET"),
+        },
+      ) do |f|
+        retry_if = ->(env, _) { env.status.nil? || env.status == 429 || env.status >= 500 }
```

Comment:
This is interesting. Could you add a comment about the `nil` and `500+` statuses, as I'm not sure why they're here, and what retrying would achieve.